Lib/pdb.py

```diff
@@ -1723,6 +1723,19 @@ def do_quit(self, arg):
 
         Quit from the debugger. The program being executed is aborted.
         """
+        if self.mode == 'inline':
+            while True:
+                try:
+                    reply = input('Quitting pdb will kill the process. Quit anyway? [y/n] ')
+                    reply = reply.lower().strip()
+                except EOFError:
+                    reply = 'y'
+                    self.message('')
+                if reply == 'y' or reply == '':
+                    os._exit(0)
```

Comment:
I think I'd prefer `sys.exit` here. `os._exit` may lead to unreleased resources. If the user wants to kill the process faster, they can hit Ctrl-C or Ctrl-\ after.

Reply:
It's always possible to have unreleased resources - we can't prevent that with `SystemExit`. It may get better in some cases, but raising `SystemExit` in an arbitrary place of the code does not seem like a very safe way to end the program to me. The only way to make sure all resources are released (if the program is written correctly) is to `continue` the program.

One of the problem of `SystemExit` is:

```python
while True:
    try:
        breakpoint()
    except:
        pass
```

This will trap in debugger forever. I know this example is a bit artificial, but it's not that rare for programs to handle `SystemExit`, and it's frustrating for users to be stuck in the debugger when they just want to quit.

We have a warning for the users already and they should be aware that they are "killing" a process - which means the resources could potentially be leaked. At least they'll know the process will definitely be killed after they say yes.

Of course that's my thought, and is open to more discussion.